frontend/src/components/MapView/Layers/AdminLevelDataLayer/index.tsx

```diff
@@ -106,6 +110,21 @@ const AdminLevelDataLayers = ({
     addFillPatternImagesInMap();
   }, [addFillPatternImagesInMap, layer.id, map]);
 
+  useEffect(() => {
+    if (!map) {
+      return () => {};
+    }
+
+    map.on('click', getLayerMapId(layer.id), onClick({ dispatch, layer, t }));
+    return () => {
+      map.off(
+        'click',
+        getLayerMapId(layer.id),
+        onClick({ dispatch, layer, t }),
+      );
+    };
+  }, [dispatch, layer, layer.id, map, t]);
```

Comment:
Let's test things a bit but if this works well let's turn this into a custom useEffect that takes a custom onClick and data as input. Then we can reuse these functions accross the different layers

You probably just need `map.off('click', layerId);` for the cleanup part. wdyt?